test/blackbox-tests/test-cases/pkg/e2e.t

```diff
@@ -0,0 +1,72 @@
+Exercises end to end locking and building a simple project.
+
+  $ . ./helpers.sh
+  $ mkrepo
+  $ add_mock_repo_if_needed
+
+Make a library:
+  $ mkdir foo
+  $ cd foo
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package (name foo))
+  > EOF
+  $ cat > foo.ml <<EOF
+  > let foo = "Hello, World!"
+  > EOF
+  $ cat > dune <<EOF
+  > (library
+  >  (public_name foo))
+  > EOF
+  $ cd ..
+  $ tar -czf foo.tar.gz foo
+  $ rm -rf foo
+
+Make a package for the library:
+  $ mkpkg foo <<EOF
+  > build: [
+  >   ["dune" "subst"] {dev}
+  >   [
+  >     "dune"
+  >     "build"
+  >     "-p"
+  >     name
+  >     "-j"
+  >     jobs
+  >     "@install"
+  >     "@runtest" {with-test}
+  >     "@doc" {with-doc}
+  >   ]
+  > ]
+  > url {
+  >  src: "file://$PWD/foo.tar.gz"
```

Comment:
Can you add the checksum as well to make sure the validation works?

Reply:
done